### PR TITLE
Extra sounds

### PR DESCRIPTION
**Pull request by bytsbyts**
sounds with various sources can play randomly ,add edited diescreaming sounds(avoided duplicate falling down sounds)

### CODE REVIEW

File: Assets/!/Scripts/PlayerAudio.cs [hunk 1]
```diff
@@ -6,52 +7,104 @@ public class PlayerAudio : MonoBehaviour
...
     public Audio[] audioPlayer;
+    public AudioClip[] audioClips;
+    private Dictionary<string, AudioClip> dicAudioPlayer = new Dictionary<string, AudioClip>();
 
+    public Dictionary<string, AudioClip> DicAudioPlayer { get => dicAudioPlayer; set => dicAudioPlayer = value; }
+
+
+    enum RandomClip
+    {
+        walkOrRun, diescream, blood
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:18]: Use PascalCase

File: Assets/!/Scripts/PlayerAudio.cs [hunk 2]
```diff
@@ -6,52 +7,104 @@ public class PlayerAudio : MonoBehaviour
...
         audioSource.loop = false;
     }
 
+
+    private string RandomAudio(string clipName, int indexClip)
+    {
+        int index = 0;
+        switch (indexClip)
+        {
+            case 0:
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:97]: Use constants, what is indexClip?

File: Assets/!/Scripts/PlayerAudio.cs [hunk 3]
```diff
@@ -6,52 +7,104 @@ public class PlayerAudio : MonoBehaviour
 {
     public AudioSource audioSource;
     public Audio[] audioPlayer;
+    public AudioClip[] audioClips;
+    private Dictionary<string, AudioClip> dicAudioPlayer = new Dictionary<string, AudioClip>();
 
+    public Dictionary<string, AudioClip> DicAudioPlayer { get => dicAudioPlayer; set => dicAudioPlayer = value; }
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:13]: These are unused

File: Assets/!/Scripts/PlayerAudio.cs [hunk 4]
```diff
@@ -6,52 +7,104 @@ public class PlayerAudio : MonoBehaviour
 {
     public AudioSource audioSource;
     public Audio[] audioPlayer;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:9]: Should have different arrays for each type of sound

File: Assets/!/Scripts/PlayerAudio.cs [hunk 5]
```diff
@@ -6,52 +7,104 @@ public class PlayerAudio : MonoBehaviour
...
+    private string RandomAudio(string clipName, int indexClip)
+    {
+        int index = 0;
+        switch (indexClip)
+        {
+            case 0:
+                index = Random.Range(1, 6);
+                break;
+            case 1:
+                index = Random.Range(1, 4);
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:101]: What if we delete one clip of this type? The game crashes -> This should be dynamic and not hardcoded

**Review by matias-gonz — CHANGES_REQUESTED** [review]

**Comment by bytsbyts** [comment]
Thanks for examination.Changed now.Please point out further mistakes

File: Assets/!/Scripts/PlayerAudio.cs [hunk 6]
```diff
@@ -29,29 +43,108 @@ public void Crawl()
...
+            audioToPlay = System.Array.Find(dieScreamAudio, a => a.id == clipName);
+        }
+        if (audioToPlay.clip == null)
+        {
+            audioToPlay = System.Array.Find(bloodAudio, a => a.id == clipName);
+        }
+        if (audioToPlay.clip == null)
+        {
+            audioToPlay = System.Array.Find(otherAudio, a => a.id == clipName);
+        }
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:114]: Defeats purpose of having different arrays, this is the same as having one big array

File: Assets/!/Scripts/PlayerAudio.cs [hunk 7]
```diff
@@ -29,29 +43,108 @@ public void Crawl()
...
+                maxIndex = bloodAudio.Length;
+                clipName = "blood";
+                break;
+        }
+
+        if (maxIndex > 0)
+        {
+            index = Random.Range(1, maxIndex + 1);
+        }
+        return clipName + index;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:148]: Should return `Audio` not `string`

File: Assets/!/Scripts/PlayerAudio.cs [hunk 8]
```diff
@@ -29,29 +43,108 @@ public void Crawl()
...
+        int index = 0;
+        int maxIndex = 0;
+        string clipName = "";
+
+        switch (clipType)
+        {
+            case RandomClip.Walk:
+                maxIndex = walkAudio.Length;
+                clipName = "walk";
+                break;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:129]:
```suggestion
                return walkAudio[Random.Range(0, maxIndex)];
```

File: Assets/!/Scripts/PlayerAudio.cs [hunk 9]
```diff
@@ -29,29 +43,108 @@ public void Crawl()
...
+        PlayAudio("FallDown", 0.7f);
+    }
+
+    private void PlayRandomAudio(RandomClip clipType, float volume)
+    {
+        string clipName = RandomAudio(clipType);
+        PlayAudio(clipName, volume);
+    }
+
     public void PlayAudio(string clipName, float volume)
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:80]: Should receive `Audio`

**Review by matias-gonz — CHANGES_REQUESTED** [review]

**Comment by bytsbyts** [comment]
Changed

File: Assets/!/Scripts/PlayerAudio.cs [hunk 10]
```diff
@@ -1,57 +1,159 @@
...
 
-
-        audioSource.clip = audioToPlay.clip;
+        audioSource.clip = audioClip.clip;
         audioSource.volume = volume;
         audioSource.Play();
         audioSource.loop = false;
     }
 
+    private Audio FindAudioClip(string clipName)
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:108]: Still didn't delete this

**Review by matias-gonz — CHANGES_REQUESTED** [review]

**Comment by bytsbyts** [comment]
Does this version work?

File: Assets/!/Scripts/PlayerAudio.cs [hunk 11]
```diff
@@ -1,57 +1,131 @@
...
 
-
-        audioSource.clip = audioToPlay.clip;
+        audioSource.clip = audioClip.clip;
         audioSource.volume = volume;
         audioSource.Play();
         audioSource.loop = false;
     }
 
+    private Audio GetSpecificAudio(string clipName, Audio[] audioArray)
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:108]: Why do you need a specific audio?

When we want to play a sound we choose randomly from the array

Comment (bytsbyts) [reply Assets/!/Scripts/PlayerAudio.cs:108]: Some sounds have only one clip, some have more so we choose randomly.In this case, could you give some advice about the different ways

Comment (bytsbyts) [reply Assets/!/Scripts/PlayerAudio.cs:108]: Or delete other array,make those with only one clip choice also individual array?

Comment (matias-gonz) [reply Assets/!/Scripts/PlayerAudio.cs:108]: If we have only one clip, the random will return always the same clip, no need to have a separate function for this case.
The problem is, what happens if you add one clip? you need to change the code

File: Assets/!/Scripts/PlayerAudio.cs (hunk 5 above)
Comment (bytsbyts) [reply Assets/!/Scripts/PlayerAudio.cs]: 截屏2024-07-21 12 59 38
Do you think this way is good?

File: Assets/!/Scripts/PlayerAudio.cs [hunk 12]
```diff
@@ -1,57 +1,100 @@
...
+    public Audio[] WalkAudio;
+    public Audio[] RunAudio;
+    public Audio[] DieScreamAudio;
+    public Audio[] BloodAudio;
+    public Audio[] LeftCrouchAudio;
+    public Audio[] RightCrouchAudio;
+    public Audio[] CrawlAudio;
+    public Audio[] BulletHitAudio;
+    public Audio[] JumpAudio;
+    public Audio[] FallDownAudio;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:18]: Serialized fields should be camelCase not PascalCase

File: Assets/!/Scripts/PlayerAudio.cs [hunk 13]
```diff
@@ -1,57 +1,100 @@
...
+            Debug.LogWarning("Audio array is empty");
             return;
         }
 
-
-        audioSource.clip = audioToPlay.clip;
-        audioSource.volume = volume;
-        audioSource.Play();
-        audioSource.loop = false;
+        Audio audioClip = audioArray.Length == 1 ? audioArray[0] : audioArray[Random.Range(0, audioArray.Length)];
```
Comment (matias-gonz) [inline Assets/!/Scripts/PlayerAudio.cs:83]: No need for the ternal: `Random.Range(0, audioArray.Length)` should return 0 if length is 1

**Review by matias-gonz — CHANGES_REQUESTED** [review]

**Comment by matias-gonz** [comment]
Close in favor of #72

**Pull request closed without merging**